src/lib.rs

```diff
@@ -405,7 +466,7 @@ where
         Ok(())
     }
 
-    fn read(&mut self, reg: Registers) -> Result<u8, Ecs, Espi> {
+    pub fn read(&mut self, reg: Registers) -> Result<u8, Ecs, Espi> {
```

Comment:
What is the reason to make this public?

Reply:
That is public because there are people that use the read function to do manual operations on the module. Keeping this private means we'd have to account for everyone's needs. Someone posted a gist that used your code, but the read function was public (I can't find the gist right now, but if you really need it I'll find the guy that I'm talking about).

Reply:
I don't mind it, I was just curious what is the motivation and also surprised that none opened an issue if it is important to have raw read access to registers.

Reply:
I'll try to follow up with more information, I'm about to go to bed. The original guy that made it public (which I copied) isn't on the Matrix channel anymore, I'll see if I can find his gist. I personally used the public read function to read the module version (I believe there was a `Version` enum in there).